third_party/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+  # Suppress warnings in third party subprojects.
+  add_compile_options(-Wno-everything)
```

Comment:
I would prefer to see all impactful settings in the top-level cmake.

Reply:
Is not easily done because the `add_compile_options(-Wno-everything)` would have to be added like this: 
```
if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
  # Suppress warnings in third party subprojects.
  add_compile_options(-Wno-everything)
endif()

add_subdirectory(third_party/onnx)
add_subdirectory(third_party/pybind11)
add_subdirectory(third_party/rapidcheck)

add_subdirectory(utils)
add_subdirectory(include)
...
```
Doing that would  affect anything that follows the `add_compile_options` and therefore it would disable warnings not just for the `third_party` code but all for the `onnx-mlir` code.

Moving
```
add_subdirectory(third_party/onnx)
add_subdirectory(third_party/pybind11)
add_subdirectory(third_party/rapidcheck)
```
to the end of the file is not possible because `cmake` will then fail to produce the makefiles.

Reply:
Also, since you're suppressing the warnings for gcc/clang, I'd like to see them suppressed for MSVC which uses a different flag and using `cxx_flags` will make that possible.